### PR TITLE
Add EuTimer to implement timeout for EuRxManager (#200)

### PR DESCRIPTION
**Pull request by zion830**
* Related Issue : #200 

I implemented `EuTimer` class for timeout by using [TimerTask](https://docs.oracle.com/javase/7/docs/api/java/util/TimerTask.html).
It can be applied like this in RxEuManager,
```
mListenThread.start();
EuTimer euTimer = new EuTimer(mListenThread, () -> { 
    mListenThread = null;
});
euTimer.start(timeout);
```
but if #217 is merged, the form of listen() can change.
So I only upload EuTimer's code to avoid conflicts
Hope you give me a feedback!

### CODE REVIEW

**Review by designe — COMMENTED** [review]
Hi! @zion830!

Naming is good at `EuTimeOutListener` & `EuTimer`.
And reusability is expected due to new class implemenation :)
However, it seems that a unit test implementation is needed for this class.
It's okay to only test against public methods.

Except unit-test, It looks great! :)

**Comment by zion830** [comment]
@designe 
Thanks for your feedback!
I added EuTimer's test code.

File: euphony/src/main/java/co/euphony/util/EuTimer.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private inner class TimeOutTask(
+        private val timer: Timer
+    ) : TimerTask() {
+
+        override fun run() {
+            if (thread.isAlive) {
+                Log.e(LOG, "Your job(${Thread.currentThread().name}) is interrupted by timeout")
+                thread.interrupt()
+                timer.cancel()
+                timeOutListener.onTimeOut()
```
Comment (phj0407) [inline euphony/src/main/java/co/euphony/util/EuTimer.kt:35]: I like detail Logs👍Test looks so good too!

**Review by yeonns — COMMENTED** [review]
@zion830 #217 doesn't change `listen()` anymore, so you don't need to consider conflict :)
Have any code to update?

**Pull request closed without merging**